### PR TITLE
Update sendMailTemplate

### PR DESCRIPTION
**Pull request by patrickvenetz**
This Pull Request enables `sendMailTemplate` to fetch an HTML template file and pass to Mandrill as opposed to passing along a template name.

It also replaces _lib/payments/sendPaymentSuccessful.js_ with a function in _lib/Mail.js_ and provides same merge variables as in pledge confirmation emails.

Template name still works as a fallback in case an HTML template file is not available.

It includes a series of HTML template files:

- pledge_abo.html
- pledge_abo_give.html
- pledge_abo_give_months.html
- pledge_benefactor.html
- pledge_prolong.html
- payment_successful_abo.html
- payment_successful_abo_give.html
- payment_successful_abo_give_months.html
- payment_successful_benefactor.html
- payment_successful_donate.html
- payment_successful_prolong.html

Templates named _payment_successful_* are new and ought to replace _cf_payment_successful_.

When environment variable `MAIL_EXPRESS_RENDER` is set to true, any HTML template file is render on path `/mail/render/:template`, `:template` representing a template string without `.html` extension.

### CODE REVIEW

File: packages/mail/express/render.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+      const html = await getTemplate(req.params.template)
+
+      if (!html) {
+        return res.sendStatus(404)
+      }
+
+      return res.send(html)
+    }
+  )
+}
```
Comment (patte) [inline packages/mail/express/render.js]: what's the purpose of this public endpoint?

File: packages/mail/lib/sendMailTemplate.js [hunk 2]
```diff
@@ -52,18 +67,24 @@ module.exports = async (mail, context, log) => {
...
     from_name: mail.fromName || DEFAULT_MAIL_FROM_NAME,
+    html: getTemplate(mail.templateName),
     merge_language: mail.mergeLanguage || 'mailchimp',
     global_merge_vars: mergeVars,
     auto_text: true,
     tags
   }
-  debug(message)
+
+  debug(_.omit(message, 'html'))
```
Comment (patte) [inline packages/mail/lib/sendMailTemplate.js]:
```suggestion
  debug({ ...message, html: !!message.html })
```

File: packages/mail/lib/sendMailTemplate.js [hunk 3]
```diff
@@ -1,7 +1,10 @@
-const debug = require('debug')('mail:lib:sendMailTemplate')
+const _ = require('lodash')
```
Comment (patte) [inline packages/mail/lib/sendMailTemplate.js]:
```suggestion
```

File: packages/mail/utils/shouldSendMessage.js [hunk 4]
```diff
@@ -14,14 +16,14 @@ module.exports = (message) => {
   // don't send in dev, expect SEND_MAILS is true
   // don't send mails if SEND_MAILS is false
   if (SEND_MAILS === 'false' || (DEV && SEND_MAILS !== 'true')) {
-    console.log('\n\nSEND_MAIL prevented mail from being sent\n(SEND_MAIL == false or NODE_ENV != production and SEND_MAIL != true):\n', message)
+    console.log('\n\nSEND_MAILS prevented mail from being sent\n(SEND_MAIL == false or NODE_ENV != production and SEND_MAIL != true):\n', _.omit(message, 'html'))
```
Comment (patte) [inline packages/mail/utils/shouldSendMessage.js]: like above, I would indicate if html is set or not.

**Comment by patte** [comment]
Which templates are provided as html and which still life on mailchimp? What's the logic here? Are you going to remove the templates existing here from mailchimp?

**Comment by patrickvenetz** [comment]
> Which templates are provided as html and which still life on mailchimp? What's the logic here? Are you going to remove the templates existing here from mailchimp?

This is sort of the proof-of-concept. Once deployed, a 2nd merge request with all remaining templates on MailChimp is coming.

File: packages/mail/express/render.js (hunk 1 above)
Comment (patrickvenetz) [reply packages/mail/express/render.js]: Soley for testing and debugging purposes.

File: packages/mail/lib/sendMailTemplate.js (hunk 2 above)
Comment (patrickvenetz) [reply packages/mail/lib/sendMailTemplate.js]: ❤️

File: packages/mail/express/render.js (hunk 1 above)
Comment (patte) [reply packages/mail/express/render.js]: ok with me

**Comment by patte** [comment]
Ok, as long as it's absolutely clear if a template is on mailchimp or in code I'm fine with it.

**Comment by patrickvenetz** [comment]
> Ok, as long as it's absolutely clear if a template is on mailchimp or in code I'm fine with it.

Once deployed, I intend to remove available templates on Mandrill and MailChimp.

**Pull request merged**